app/controllers/comments_controller.rb

```diff
@@ -9,14 +10,23 @@ def create
     @comment = Comment.new(comment_params)
     @comment.user = current_user
     if @comment.save
-      redirect_to post_path(@comment.post_id)
+      if params[:comment_id]
+        redirect_to root_path
+      else
+        redirect_to post_path(@commentable)
+      end
     else
       redirect_back(fallback_location: root_path)
     end
   end
 
   private
     def comment_params
-      params.require(:comment).permit(:body, :post_id, :comment_id)
+      params.require(:comment).permit(:body, :commentable_type, :commentable_id)
+    end
+
+    def find_commentable
+      @commentable = Comment.find_by_id(params[:comment_id]) if params[:comment_id]
```

Comment:
If both a comment and post id are present the post overrides the comment, is that correct?

Reply:
I thought only a comment id or a post id would be present, not both. Isn't it deriving this from commentable_id?

Reply:
I don't think you can assume that in all cases. The params are user input, so they could both be present even if your app code only ever passes in one. (It might not be bad to default to the Post, just something to think about.)